client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -73,15 +79,28 @@ public int getAppendBlockSize() {
         if (numInflight <= 1) {
             return 0;
         }
-        double appendsInMaxBatch = Math.max(1.0, MAX_BATCH_TIME_MILLIS / millisBetweenAppends.getCurrentValue());
-        double targetAppendsOutstanding = MathHelpers.minMax(appendsOutstanding.getCurrentValue() * 0.5, 1.0,
-                                                             appendsInMaxBatch);
-        return (int) MathHelpers.minMax((long) (targetAppendsOutstanding * eventSize.getCurrentValue()), 0,
-                                        MAX_BATCH_SIZE);
+        double nanosPerAppend = nanosBetweenAppends.getCurrentValue();
+        double appendsInMaxBatchTime = (MAX_BATCH_TIME_MILLIS * NANOS_PER_MILLI) / nanosPerAppend;
+        double appendsInTime = Math.max(1.0, BASE_TIME_NANOS / nanosPerAppend);
+        double appendsInBatch = MathHelpers.minMax(appendsOutstanding.getCurrentValue() * OUTSTANDING_FRACTION, appendsInTime, appendsInMaxBatchTime);
+        int size = (int) (appendsInBatch * eventSize.getCurrentValue()) + BASE_SIZE;
+        return MathHelpers.minMax(size, 0, MAX_BATCH_SIZE);
     }
 
     @Override
     public int getBatchTimeout() {
         return MAX_BATCH_TIME_MILLIS;
     }
+    
+    private static int initializeConstant(String name, int defaultValue) {
+        String val = System.getenv(name);
```

Comment:
is is a common practice in Pravega to use System env for the client ?

Here we are in the client, we don't know much about the running environment of the client application.

Additionally using ENV may be seen as a security issue, as the application is going to use parameters configured outside the configuration files of the application itself, without even noticing it.

We should at least log a line telling that we picked up a value from system ENV

Reply:
These are "hardcoded" constants which we do not want users to set (we will not document them). The point of making them ENV vars is because we want to try out various combinations in our perf tests without having to do a full rebuild.

Reply:
@andreipaduroiu 
What about adding a system property to enable this behaviour ?
`-Dpravega.allowreadconstantsfromenv=true`

This way we will protect clients from running into (security) issues.
System properties are not read automatically from the ENV or by the JVM, so you have to explicitly opt-in to this way of configuring these parameters.

Reply:
a system property approached is used here. ref: https://github.com/pravega/pravega/pull/4873/files#diff-642bfb8d9badad7de4a71c8d06a9a9baR67 io.pravega.client.connection.impl.SocketConnectionFactoryImpl#getThreadPoolSize

Reply:
@RaulGracia Would you prefer prefer these to: a) stay as-is. b) change into system properties
or are we done with testing these and we can go back to hardcoding them?